lib/modules/manager/helmv3/common.ts

```diff
@@ -1,19 +1,40 @@
 import { quote } from 'shlex';
 import upath from 'upath';
 
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
 import type { ExtraEnv } from '../../../util/exec/types';
 import { privateCacheDir } from '../../../util/fs';
+import { addSecretForSanitizing } from '../../../util/sanitize';
+import { fromBase64 } from '../../../util/string';
+import { ecrRegex, getECRAuthToken } from '../../datasource/docker/ecr';
 import type { RepositoryRule } from './types';
 
-export function generateLoginCmd(
+export async function generateLoginCmd(
   repositoryRule: RepositoryRule,
   loginCMD: string
-): string | null {
-  const { username, password } = repositoryRule.hostRule;
-  if (username && password) {
+): Promise<string | null> {
+  const { hostRule, repository } = repositoryRule;
+  const { username, password } = hostRule;
+  if (ecrRegex.test(repository)) {
+    logger.trace({ repository }, `Using ecr auth for Helm registry`);
+    const [, region] = coerceArray(ecrRegex.exec(repository));
+    const auth = await getECRAuthToken(region, hostRule);
+    if (auth) {
```

Comment:
```suggestion
    if (!auth) {
      return null
    }
```
Reverse for easier readability and less indentation.

Reply:
Sounds good to me, but this does actually change the behaviour back to a previous behaviour from before applying a suggestion by @viceice - could I get some clarification on the desired behaviour?

There are basically 2 options I think:

1. When ECR auth fails, fall back to trying the original username + password as regular creds.
2. When ECR auth fails, don't try to log in at all.

For reference, it seems like the `docker` datasource currently does option 2.